storage/sql/src/test/java/io/apicurio/registry/storage/impl/sql/SqlRegistryStorageTest.java

```diff
@@ -39,4 +68,238 @@ protected RegistryStorage storage() {
+        tenantCtx.tenantId(tenantId1);
+        searchResults = storage().searchArtifacts("testMultiTenant_Search", 0, 100, SearchOver.labels, SortOrder.asc);
+        Assertions.assertNotNull(searchResults);
+        Assertions.assertEquals(10, searchResults.getCount());
+        Assertions.assertEquals(10, searchResults.getArtifacts().size());
+    }
+
+    @Test
+    public void testMultiTenant_GlobalRules() {
+        tenantCtx.tenantId(tenantId1);
+
+        List<RuleType> globalRules = storage().getGlobalRules();
+        Assertions.assertNotNull(globalRules);
+        Assertions.assertTrue(globalRules.isEmpty());
+        
+        RuleConfigurationDto config = new RuleConfigurationDto();
+        config.setConfiguration("FULL");
+        storage().createGlobalRule(RuleType.COMPATIBILITY, config);
+        try {
+            storage().createGlobalRule(RuleType.COMPATIBILITY, config);
+            Assertions.fail("Expected a RuleAlreadyExistsException for " + RuleType.COMPATIBILITY);
+        } catch (RuleAlreadyExistsException e) {
+            // this is expected
+        }
+        
+        RuleConfigurationDto rule = storage().getGlobalRule(RuleType.COMPATIBILITY);
+        Assertions.assertEquals(rule.getConfiguration(), config.getConfiguration());
+        
+        globalRules = storage().getGlobalRules();
+        Assertions.assertNotNull(globalRules);
+        Assertions.assertFalse(globalRules.isEmpty());
+        Assertions.assertEquals(globalRules.size(), 1);
+        Assertions.assertEquals(globalRules.get(0), RuleType.COMPATIBILITY);
+        
+        ///////////////////////////////////////////////////
+        // Now switch to tenant #2 and do the same stuff
+        ///////////////////////////////////////////////////
+        
+        tenantCtx.tenantId(tenantId2);
+        globalRules = storage().getGlobalRules();
+        Assertions.assertNotNull(globalRules);
+        Assertions.assertTrue(globalRules.isEmpty());
+        
+        config = new RuleConfigurationDto();
+        config.setConfiguration("FULL");
+        storage().createGlobalRule(RuleType.COMPATIBILITY, config);
+        try {
+            storage().createGlobalRule(RuleType.COMPATIBILITY, config);
+            Assertions.fail("Expected a RuleAlreadyExistsException for " + RuleType.COMPATIBILITY);
+        } catch (RuleAlreadyExistsException e) {
+            // this is expected
+        }
+        
+        rule = storage().getGlobalRule(RuleType.COMPATIBILITY);
+        Assertions.assertEquals(rule.getConfiguration(), config.getConfiguration());
+        
+        globalRules = storage().getGlobalRules();
+        Assertions.assertNotNull(globalRules);
+        Assertions.assertFalse(globalRules.isEmpty());
+        Assertions.assertEquals(globalRules.size(), 1);
+        Assertions.assertEquals(globalRules.get(0), RuleType.COMPATIBILITY);
+    }
+    
+    public void testMultiTenant_ArtifactNotFound() throws Exception {
```

Comment:
missing `@Test` here and in all methods below?

Reply:
whoops!  I knew it was too good to be true...